source/extensions/filters/network/dubbo_proxy/router/router_impl.cc

```diff
@@ -221,6 +221,32 @@ void Router::onEvent(Network::ConnectionEvent event) {
   }
 }
 
+const Envoy::Router::MetadataMatchCriteria* Router::metadataMatchCriteria() {
+
+  const Envoy::Router::MetadataMatchCriteria* route_criteria =
+      (route_entry_ != nullptr) ? route_entry_->metadataMatchCriteria() : nullptr;
+  // have we been called before? If so,there's no need to recompute because
+  // by the time this method is called for the first time,route_entry_ should
+  // not change anymore
+  if (metadata_match_ != nullptr) {
+    return metadata_match_.get();
+  }
+  // The request's metadata,if present,takes precedence over the route's
```

Comment:
Mind spaces after commas and dots after sentences' ends.
```suggestion
  // The request's metadata, if present, takes precedence over the route's.
```